state/testing/watcher.go

```diff
@@ -157,7 +157,13 @@ func (c StringsWatcherC) AssertChanges() {
 }
 
 func (c StringsWatcherC) AssertChange(expect ...string) {
-	c.assertChange(false, expect...)
+	// Fixes lp#1589641: some time, under race & stress testing,
+	// reads of changes from watcher chan seem to get out of order.
+	// This additional Sync, ensures that the changes are processed correctly.
+	c.State.StartSync()
```

Comment:
This seems overbroad, as things should be doing some sort of StartSync in the function themselves, rather than as a side effect of AssertChange.
Is there a specific test which is failing? Or is this a group of tests that are all missing the synchronization?